src/app/components/fields/sd-text/sd-text.component.ts

```diff
@@ -1,82 +1,61 @@
-import { Component, OnInit, Input, OnChanges, SimpleChange, SimpleChanges, DoCheck } from "@angular/core";
-import { Observable } from "rxjs";
+import { Component, OnInit, Input, Output, EventEmitter } from "@angular/core";
 
-const COMMIT_DEBOUNCE: number = 250;
 
 @Component({
   selector: "app-sd-text",
   templateUrl: "./sd-text.component.html",
   styleUrls: ["./sd-text.component.less"]
 })
 export class SdTextComponent implements OnInit {
-  @Input() target: any;
+  @Input() required: boolean; // TODO: Есть в HTML, но почему-то нет в ctrl
   @Input() minlength: number;
   @Input() maxlength: number;
   @Input() allowEmpty: boolean;
   @Input() editing: boolean;
   @Input() placeholder: string;
   @Input() emptyValue: string;
-  @Input() validate: any;
   @Input() disabled: boolean;
+  @Input() validate: (value: string) => string;
 
-  value: any;
-  commitTask: any;
   enabled: boolean;
-  validationError: any;
+  errorMessage: any;
 
-  constructor() {}
+  private _value: string;
+  @Output() valueChange = new EventEmitter<string>();
+
+  constructor() { }
 
   ngOnInit() {
-    // TODO: Разобраться зачем ниже описанный код
-    // this.value = this.target;
-    // this.$scope.$watch("ctrl.editing", () => {
```

Comment:
вроде все ок.. осталось только кейсы у компонентов проверить
то есть изменение editing, вызов внешней функции при изменении значения и валидацию

Reply:
и чтобы не спамить emit-ами onChange, добавить задержку

Reply:
как это все будет работать надо бы почистить код от мусора

Reply:
смена editing - работает
вызов валидации - работает
вызов внешней функции - работает 
```suggestion
    <app-sd-text
        [value]="editable"
        
      > (valueChange)="test($event)"

        [editing]="firstToggler"
        [validate]="validate">
    </app-sd-text>
```
Сама функция

```suggestion
    test(event: string): void {
        this.editable = event;
        console.log(event);
    }
```